cmake/BuildLibCURL.cmake

```diff
@@ -151,6 +153,9 @@ macro(curl_build)
     list(APPEND LIBCURL_CMAKE_FLAGS "-DCURL_DISABLE_PROXY=OFF")
     list(APPEND LIBCURL_CMAKE_FLAGS "-DENABLE_IPV6=ON")
     list(APPEND LIBCURL_CMAKE_FLAGS "-DCURL_DISABLE_COOKIES=OFF")
+    list(APPEND LIBCURL_CMAKE_FLAGS "-DBUILD_EXAMPLES=ON")
```

Comment:
1. The option just enables the make target and don't actually build examples. Anyway, our goal here is to build a static library and headers. We don't need the target to build examples. It seems meaningful to set this option to `OFF` in our build script.
2. The commit hash is too short and ambiguous:

   ```
   $ git show dfdd97
   error: short object ID dfdd97 is ambiguous
   hint: The candidates are:
   hint:   dfdd978f7 commit 2024-04-28 - cmake: add `BUILD_EXAMPLES` option to build examples
   hint:   dfdd979e0 blob
   ```

   An ancient default is 7 hex digits, but it seems not enough for large repositories now (see git 2.11 [release notes](https://github.com/git/git/blob/e9356ba3ea2a6754281ff7697b3e5a1697b21e24/Documentation/RelNotes/2.11.0.txt#L22-L26)). I would use 12 digits when unsure.
3. The option was initially set to `OFF` by default in 8.8.0, but then the default was switched to `ON` in https://github.com/curl/curl/commit/45202cbba4bb3d12b4469063864b57d2f8765d9c (it is in current `master`, not released yet). The commit message says nothing, whether the options are set to its curl's default, so I just noted it here.

Reply:
Turned OFF `BUILD_EXAMPLES` option and replaced all 6 hex digits commit hashs with 12 hex digit hashs. Rephrased changelog sentence.